internal/core/common_config/main.go

```diff
@@ -56,23 +66,62 @@ func Main(ctx context.Context, cancel context.CancelFunc) {
 			return lc
 		},
 	})
-
 	secretProvider, err := secret.NewSecretProvider(nil, environment.NewVariables(lc), ctx, startupTimer, dic, common.CoreCommonConfigServiceKey)
 	if err != nil {
 		lc.Errorf("failed to create Secret Provider: %v", err)
 		os.Exit(1)
 	}
 
 	lc.Info("Secret Provider created")
+	var accessToken string
+	var getAccessToken types.GetAccessTokenCallback
+
+	if secretProvider != nil {
+		getAccessToken = func() (string, error) {
+			accessToken, err = secretProvider.GetAccessToken("consul", common.CoreCommonConfigServiceKey)
+			if err != nil {
+				return "", fmt.Errorf("failed to get Configuration Provider access token: %s", err.Error())
+			}
+			return accessToken, err
+		}
+		_, err = getAccessToken()
+		if err != nil {
+			lc.Errorf("failed to get Access Token for config provider: %s", err.Error())
+			os.Exit(1)
+		}
+	}
 
-	_, err = secretProvider.GetAccessToken("consul", common.CoreCommonConfigServiceKey)
+	lc.Info("Got Config Provider Access Token")
```

Comment:
move this logging inside the `if secretProvider != nil`  block once you successfully got token (i.e. after line 91)

Reply:
If we move this to inside the callback, it gets printed twice because we make the callback right after to do the error handling.

Reply:
never mind, since `if secretProvider != nil` block is gone, this becomes irrelevant